clang_tidy.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+function usage() {
+    echo >&2 "Usage: $0 <directory containing compile_commands.json>"
+    echo >&2 "Typical usage: make sure CMake has been run with -DCMAKE_EXPORT_COMPILE_COMMANDS=1, then use:"
```

Comment:
Why not just enable this flag by default. The overhead is extremely minimal, and it simplifies this.

Reply:
Apparently you already did this in 8ba731e9, so that simplifies this :)